Kv is leader

mmlb opened the pull request:
#### Description:

KV provider task to query if this node is running on the kv leader node.

Missing some tests for to verify the false case is correct.
#### Issues affected:

Resolves #204

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cerana/cerana/362)

<!-- Reviewable:end -->

mmlb committed:
pkg/kv/consul: add IsLeader

mmlb committed:
pkg/kv/etcd: add IsLeader

mmlb committed:
pkg/kv: add IsLeader to KV interface

mmlb committed:
providers/kv: add kv-is-leader task

bbickerton commented:
Reviewed 6 of 8 files at r1.
Review status: 6 of 8 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/cerana/cerana/362)_

<!-- Sent from Reviewable.io -->

mmlb committed:
pkg/kv: tests: pass KVURL instead of just port to get

mmlb committed:
pkg/kv: tests: run tests with a 3 node cluster

Each cluster member is bound to a different IP in the IPv4 localhost
subnet so that all non client ports can stay at default.

bbickerton commented:
Reviewed 4 of 6 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

_[pkg/kv/consul/consul.go, line 320 [r2]](https://reviewable.io:443/reviews/cerana/cerana/362#-KQWmDbTdmBWeYMvBKEK:-KQWmDbTdmBWeYMvBKEL:btcpylx) ([raw file](https://github.com/cerana/cerana/blob/e8ded8644e442ee8caa6ac21f7cfb5db40fc2a6a/pkg/kv/consul/consul.go#L320)):_

> ``` Go
>   leader, _, err := net.SplitHostPort(leaderAddr)
>   if err != nil {
>       return false, errors.Wrap(err)
> ```

Might be more informative `errors.Wrapv(err, map[string]interface{}{"addr":leaderAddr}`

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/cerana/cerana/362)_

<!-- Sent from Reviewable.io -->
